android/build.gradle

```diff
@@ -4,14 +4,41 @@ apply from: './jacoco.gradle'
 apply from: './native.gradle'
 apply from: './sourcemaps.gradle'
 
+/* Checking APG version to be backward-compatible with RN versions < 0.71 */
+def agpVersion = com.android.Version.ANDROID_GRADLE_PLUGIN_VERSION.tokenize('.')[0].toInteger()
+def shouldUseNameSpace = agpVersion >= 7
+def PACKAGE_PROP = "package=\"com.instabug.reactlibrary\""
+def manifestOutFile = file("${projectDir}/src/main/AndroidManifest.xml")
+def manifestContent = manifestOutFile.getText()
+
 String getExtOrDefault(String name) {
     if (rootProject.ext.has(name)) {
         return rootProject.ext.get(name)
     }
     return project.properties['InstabugReactNative_' + name]
 }
 
+if(shouldUseNameSpace){
+      manifestContent = manifestContent.replaceAll(
+        PACKAGE_PROP,
+        ''
+    )  
+} else {
+    if(!manifestContent.contains("$PACKAGE_PROP")){
```

Comment:
How about we add an `isPackageNamespaceMissing` instead of `!manifestContent.contains("$PACKAGE_PROP"` property? It would make the logic more transparent and help us understand the condition better.

Reply:
Agree with that, I will push new update.